benchmarks/elementary_functions_experiments_benchmark.cpp

```diff
@@ -237,10 +251,22 @@ Value TableSpacingImplementation<table_spacing>::Sin(Argument const x) {
   auto const h = x - x₀;
   auto const h² = h * h;
   auto const h³ = h² * h;
-  auto const sin_x₀_plus_h_cos_x₀ = TwoProductAdd(cos_x₀, h, sin_x₀);
-  return sin_x₀_plus_h_cos_x₀.value +
-         ((sin_x₀ * h² * CosPolynomial(h²) + cos_x₀ * h³ * SinPolynomial(h²)) +
+
+#define PRINCIPIA_SIN_IMPLEMENTATION(fma_policy)            \
+  auto const sin_x₀_plus_h_cos_x₀ =                         \
+      TwoProductAdd<(fma_policy)>(cos_x₀, h, sin_x₀);       \
+  return sin_x₀_plus_h_cos_x₀.value +                       \
+         ((sin_x₀ * h² * CosPolynomial<(fma_policy)>(h²) +  \
+           cos_x₀ * h³ * SinPolynomial<(fma_policy)>(h²)) + \
           sin_x₀_plus_h_cos_x₀.error);
+
+  if (UseHardwareFMA) {
+    PRINCIPIA_SIN_IMPLEMENTATION(FMAPolicy::Force)
+  } else {
+    PRINCIPIA_SIN_IMPLEMENTATION(FMAPolicy::Disallow)
+  }
```

Comment:
Instead of wrapping half the implementation in a macro, templatize the whole function Sin or Cos on FMAPolicy, and have a non-templatized Sin or Cos call UseHardwareFMA ? Sin\<FMAPolicy::Force> : Sin\<FMAPolicy::Disallow>.